src/script/opcodes/flow_control.zig

```diff
@@ -0,0 +1,202 @@
+test "OP_IF - false condition" {
+    const allocator = testing.allocator;
+
+    const script_bytes = [_]u8{ Opcode.OP_0.toBytes(), Opcode.OP_IF.toBytes() };
+    const script = Script.init(&script_bytes);
+
+    var engine = Engine.init(allocator, script, ScriptFlags{});
+    defer engine.deinit();
+
+    try engine.execute();
+
+    try testing.expectEqual(1, engine.cond_stack.len());
+    try testing.expect(!engine.cond_stack.isBranchExecuting());
+}
+
+// Test OP_NOTIF with a true condition (OP_1)
+// Expect: Conditional stack has one item and branch is not executing (inverted)
+test "OP_NOTIF - true condition" {
+    const allocator = std.testing.allocator;
+
+    const script_bytes = [_]u8{ Opcode.OP_1.toBytes(), Opcode.OP_NOTIF.toBytes() };
+    const script = Script.init(&script_bytes);
+
+    var engine = Engine.init(allocator, script, ScriptFlags{});
+    defer engine.deinit();
+
+    try engine.execute();
+
+    try std.testing.expectEqual(1, engine.cond_stack.len());
+    try std.testing.expect(!engine.cond_stack.isBranchExecuting());
+}
+
+// Test OP_NOTIF with a false condition (OP_0)
+// Expect: Conditional stack has one item and branch is executing (inverted)
+test "OP_NOTIF - false condition" {
+    const allocator = std.testing.allocator;
+
+    const script_bytes = [_]u8{ Opcode.OP_0.toBytes(), Opcode.OP_NOTIF.toBytes() };
+    const script = Script.init(&script_bytes);
+
+    var engine = Engine.init(allocator, script, ScriptFlags{});
+    defer engine.deinit();
+
+    try engine.execute();
+
+    try std.testing.expectEqual(1, engine.cond_stack.len());
+    try std.testing.expect(engine.cond_stack.isBranchExecuting());
+}
+
+test "OP_ELSE" {
+    const allocator = std.testing.allocator;
+
+    // Test OP_ELSE with a matching OP_IF
+    // Expect: Conditional stack state is toggled correctly
+    {
+        const script_bytes = [_]u8{ Opcode.OP_IF.toBytes(), Opcode.OP_ELSE.toBytes() };
+        const script = Script.init(&script_bytes);
+        var engine = Engine.init(allocator, script, ScriptFlags{});
+        defer engine.deinit();
+
+        try opIf(&engine);
+        try opElse(&engine);
+        try std.testing.expectEqual(1, engine.cond_stack.len());
+    }
+
+    // Test OP_ELSE without a matching OP_IF
+    // Expect: Error due to empty conditional stack
+    {
+        const script_bytes = [_]u8{Opcode.OP_ELSE.toBytes()};
+        const script = Script.init(&script_bytes);
+        var engine = Engine.init(allocator, script, ScriptFlags{});
+        defer engine.deinit();
+
+        try std.testing.expectError(ConditionalStackError.UnbalancedConditional, opElse(&engine));
+    }
+}
+
+// Test OP_ENDIF with a matching OP_IF
+// Expect: Conditional stack is empty after OP_ENDIF
+test "OP_ENDIF" {
+    const allocator = std.testing.allocator;
+
+    // Test OP_ENDIF with matching OP_IF
+    {
+        const script_bytes = [_]u8{ Opcode.OP_IF.toBytes(), Opcode.OP_ENDIF.toBytes() };
```

Comment:
You don't need to run opcodes functions directly to test it, you can test OP_ENDIF like this:

```
 const script_bytes = [_]u8{
   Opcode.OP_0.toBytes(),
   Opcode.OP_IF.toBytes(),
   Opcode.OP_1.toBytes(),
   Opcode.OP_ENDIF.toBytes(),
};
```

What we expect:
```
engine.stack.len() == 0
```